main/src/library/Fixpoint/Interpreter.flix

```diff
@@ -0,0 +1,210 @@
+                }
+            case BoolExp.Eq(lhs, rhs) =>
+                evalTerm(env, lhs) == evalTerm(env, rhs)
+            case BoolExp.Leq(leq, lhs, rhs) =>
+                evalTerm(env, lhs) `leq` evalTerm(env, rhs)
+            case BoolExp.Guard0(f) =>
+                f()
+            case BoolExp.Guard1(f, t) =>
+                let v = evalTerm(env, t);
+                f(v)
+            case BoolExp.Guard2(f, t1, t2) =>
+                let v1 = evalTerm(env, t1);
+                let v2 = evalTerm(env, t2);
+                f(v1)(v2)
+            case BoolExp.Guard3(f, t1, t2, t3) =>
+                let v1 = evalTerm(env, t1);
+                let v2 = evalTerm(env, t2);
+                let v3 = evalTerm(env, t3);
+                f(v1)(v2)(v3)
+            case BoolExp.Guard4(f, t1, t2, t3, t4) =>
+                let v1 = evalTerm(env, t1);
+                let v2 = evalTerm(env, t2);
+                let v3 = evalTerm(env, t3);
+                let v4 = evalTerm(env, t4);
+                f(v1)(v2)(v3)(v4)
+            case BoolExp.Guard5(f, t1, t2, t3, t4, t5) =>
+                let v1 = evalTerm(env, t1);
+                let v2 = evalTerm(env, t2);
+                let v3 = evalTerm(env, t3);
+                let v4 = evalTerm(env, t4);
+                let v5 = evalTerm(env, t5);
+                f(v1)(v2)(v3)(v4)(v5)
+        } as & Pure, es)
+
+    def evalTerm(env: SearchEnv[v], term: RamTerm[v]): v with ToString[v] = match term {
+        case RamTerm.Lit(v) => v
+        case RamTerm.Attribute(RowVar.Index(i), index) =>
+            let (tupleEnv, _) = env;
+            tupleEnv[i][index] as & Pure
+        case RamTerm.Lat(RowVar.Index(i)) =>
+            let (_, latEnv) = env;
+            latEnv[i] as & Pure
+        case RamTerm.Meet(cap, lhs, rhs) =>
+            evalTerm(env, lhs) `cap` evalTerm(env, rhs)
+        case RamTerm.App0(f) =>
+            f()
+        case RamTerm.App1(f, t) =>
+            let v = evalTerm(env, t);
+            f(v)
+        case RamTerm.App2(f, t1, t2) =>
+            let v1 = evalTerm(env, t1);
+            let v2 = evalTerm(env, t2);
+            f(v1)(v2)
+        case RamTerm.App3(f, t1, t2, t3) =>
+            let v1 = evalTerm(env, t1);
+            let v2 = evalTerm(env, t2);
+            let v3 = evalTerm(env, t3);
+            f(v1)(v2)(v3)
+        case RamTerm.App4(f, t1, t2, t3, t4) =>
+            let v1 = evalTerm(env, t1);
+            let v2 = evalTerm(env, t2);
+            let v3 = evalTerm(env, t3);
+            let v4 = evalTerm(env, t4);
+            f(v1)(v2)(v3)(v4)
+        case RamTerm.App5(f, t1, t2, t3, t4, t5) =>
+            let v1 = evalTerm(env, t1);
+            let v2 = evalTerm(env, t2);
+            let v3 = evalTerm(env, t3);
+            let v4 = evalTerm(env, t4);
+            let v5 = evalTerm(env, t5);
+            f(v1)(v2)(v3)(v4)(v5)
+        case _ => bug!("Illegal term ${term}")
```

Comment:
How can this happen? Is this be cause the pattern matcher is broken or something else?

Reply:
It's because my Lowering phase is supposed to get rid of all named row variables and replace them with indices. Maybe that case should be explicit.